client/app/bundles/course/survey/containers/SurveyLayout/index.jsx

```diff
@@ -1,141 +1,63 @@
+        : {};
+
+    const page = Component.displayName;
+
+    return (
+      <main className="space-y-5">
+        {survey && (
+          <PageHeader
+            returnLink={backLocations(courseId, surveyId, page)}
+            title={survey.title}
+            toolbars={
+              surveyId && [
+                <AdminMenu key="admin-menu" {...{ survey, surveyId }} />,
+              ]
+            }
+          />
+        )}
 
-  return (
-    <main className="space-y-5">
-      {survey && (
-        <PageHeader
-          returnLink={backLocations(courseId, surveyId, page)}
-          title={survey.title}
-          toolbars={
-            surveyId
-              ? [<AdminMenu key="admin-menu" {...{ survey, surveyId }} />]
-              : null
-          }
-        />
-      )}
-      {pageToRender}
-    </main>
-  );
-};
+        <Component courseId={courseId} survey={survey} surveyId={surveyId} />
 
-PageWithTitleBar.propTypes = {
-  page: PropTypes.string,
-  survey: surveyShape,
-  surveyId: PropTypes.string,
-  courseId: PropTypes.string,
-};
+        <Dialogs />
+      </main>
+    );
+  };
 
-const SurveyLayout = ({ surveys }) => {
-  const params = useParams();
-  const surveyId = params.surveyId;
-  const courseId = params.courseId;
-  const survey =
-    surveys && surveys.length > 0
-      ? surveys.find((s) => String(s.id) === String(params.surveyId))
-      : {};
+  WrappedComponent.propTypes = { surveys: PropTypes.arrayOf(surveyShape) };
+  WrappedComponent.displayName = `withSurveyLayout(${Component.displayName})`;
 
-  return (
-    <Routes>
-      <Route
-        element={
-          <PageWithTitleBar
-            {...{ page: 'SurveyShow', survey, courseId, surveyId }}
-          />
-        }
-        path=""
-      />
-      <Route
-        element={
-          <PageWithTitleBar
-            {...{ page: 'SurveyResults', survey, courseId, surveyId }}
-          />
-        }
-        exact
-        path="/results"
-      />
-      <Route
-        element={
-          <PageWithTitleBar
-            {...{ page: 'ResponseIndex', survey, courseId, surveyId }}
-          />
-        }
-        exact
-        path="/responses"
-      />
-      <Route
-        element={
-          <PageWithTitleBar
-            {...{ page: 'ResponseShow', survey, courseId, surveyId }}
-          />
-        }
-        exact
-        path="/responses/:responseId"
-      />
-      <Route
-        element={
-          <PageWithTitleBar
-            {...{ page: 'ResponseEdit', survey, courseId, surveyId }}
-          />
-        }
-        exact
-        path="/responses/:responseId/edit"
-      />
-    </Routes>
+  return connect(({ surveys }) => ({ surveys: surveys.surveys }))(
+    WrappedComponent,
   );
 };
 
-SurveyLayout.propTypes = {
-  surveys: PropTypes.arrayOf(surveyShape),
-};
-
-export default connect((state) => ({ surveys: state.surveys }))(SurveyLayout);
+export default withSurveyLayout;
```

Comment:
Any reason why this is a HOC instead of being used as a component at `:surveyId` route level?

Reply:
Because `ResponseEdit`, `ResponseIndex`, `ResponseShow`, `SurveyResults`, and `SurveyShow` all need `courseId`, `survey`, and `surveyId` as props. `withSurveyLayout` provides these for them so that they can be declared in `RoutedApp` as `<ResponseEdit />`. See line 48 above.